### ISSUE TITLE
Tests fail when toolchain defaults to PIE

### ISSUE THREAD

**Issue by mwhudson** [constabulary/gb#607]
As you can see on https://launchpadlibrarian.net/263962603/buildlog_ubuntu-yakkety-amd64.gb_0.4.1-1_BUILDING.txt.gz

This was fixed in Go by https://go-review.googlesource.com/18359. I guess you need something similarly gross for gb.

**Comment by winduptoy** [comment]
This is also screwing me when trying to compile https://github.com/mattn/go-sqlite3.

I can't really decipher what they did in that patch, but I'm guessing it's not something I can solve myself by passing flags to `gb build -ldflags` or `-gcflags`?

**Comment by davecheney** [comment]
Can you please post some details, including the gb env and go env.

**Comment by winduptoy** [comment]
Ubuntu 16.10.

This doc may be of some help: https://wiki.ubuntu.com/SecurityTeam/PIE

`gb env`:
```
GB_PROJECT_DIR="/home/matt/dev/fitness-liberator"
GB_SRC_PATH="/home/matt/dev/fitness-liberator/src:/home/matt/dev/fitness-liberator/vendor/src"
GB_PKG_DIR="/home/matt/dev/fitness-liberator/pkg/linux-amd64"
GB_BIN_SUFFIX="-linux-amd64"
GB_GOROOT="/usr/local/go"
```

`go env`:
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/matt/.go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build847199824=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

**Comment by davecheney** [comment]
Does this only happen with the version of Go that Ubuntu ships?

**Comment by winduptoy** [comment]
@davecheney Not sure. I'm using go 1.7.1 linux/amd64 downloaded directly from https://golang.org

**Comment by winduptoy** [comment]
Updated to go 1.7.3, problem persists.

Tried to install @rdwilliamson's fork but I get this error when using `go get github.com/rdwilliamson/gb/...`:

```
package github.com/rdwilliamson/gb
	imports github.com/constabulary/gb/internal/debug: use of internal package not allowed
...
```

@davecheney how can I pull the @rdwilliamson fork to try it out?

Also to note, this doesn't break on every package, only some of them. Here's some I've found so far:

- https://github.com/mattn/go-sqlite3
- https://github.com/veandco/go-sdl2

**Comment by rdwilliamson** [comment]
```
cd $GOPATH/src/github.com/constabulary/gb
git remote add rdwilliamson https://github.com/rdwilliamson/gb.git
git fetch rdwilliamson
git cherry-pick rdwilliamson/master
go install github.com/constabulary/gb/...
```

As a note, I'm only seeing this problem on my Debian testing/stretch machine and not the stable/jessie ones.

**Comment by mwhudson** [comment]
> As a note, I'm only seeing this problem on my Debian testing/stretch machine and not the stable/jessie ones.

That would be because PIE is only the default in stretch, not jessie.

**Comment by davecheney** [comment]
Fixed by #655
